modules/system/helpers/ModelFinder.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace System\Helpers;
+
+use Winter\Storm\Support\Facades\File;
+use Illuminate\Support\Str;
+use Symfony\Component\Finder\Finder;
+
+class ModelFinder
+{
+    /**
+     * Find all models in core and active plugins.
+     *
+     * @return Collection
+     */
+    public static function findModels(): array
+    {
+        $models = [];
+        $models[] = static::findCoreModels();
+        $models[] = static::findActivePluginsModels();
+
+        return collect($models)->flatten()->all();
```

Comment:
Are we doing any sort of checking to make sure the classes are actual model classes?

Reply:
Not really, no.

Reply:
@LukeTowers But I do search only under `/models/` folders

Reply:
> Are we doing any sort of checking to make sure the classes are actual model classes?

Any suggestions ?

Reply:
@mjauvin if the intention is to find out if it's a model *without* loading it, you'll have to use something like [PHP Parser](https://github.com/nikic/php-parser) to statically analyze the file.

Reply:
@bennothommo I think searching for all class files under `models` is good enough for the purpose, more logic can be added by the code using the ModelFinder helper once it gets the list of class files as this is more efficient.